core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckService.java

```diff
@@ -155,11 +158,16 @@ public static HealthCheckServiceBuilder builder() {
                        AggregatedHttpResponse healthyResponse, AggregatedHttpResponse unhealthyResponse,
                        long maxLongPollingTimeoutMillis, double longPollingTimeoutJitterRate,
                        long pingIntervalMillis, @Nullable HealthCheckUpdateHandler updateHandler,
+                       Iterable<HealthCheckUpdateListener> updateListeners, boolean startHealthy,
                        Set<TransientServiceOption> transientServiceOptions) {
         serverHealth = new SettableHealthChecker(false);
+        if (!Iterables.isEmpty(updateListeners)) {
+            addServerHealthUpdateListener(ImmutableList.copyOf(updateListeners));
+        }
```

Comment:
How about taking `List<HealthCheckUpdateListener>` instead? We don't need a high level abstraction for internal implementation for which optimized code is better. For consitency, please change `Iterable<HealthChecker>` to `List<HealthChecker>`. 

```suggestion
        if (!updateListeners.isEmpty()) {
            addServerHealthUpdateListener(updateListeners);
        }
```

Reply:
Good !